src/main/java/ws/wamp/jawampa/WampClient.java

```diff
@@ -367,6 +378,13 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
                         featuresNode.put("pattern_based_subscription", true);
                     }
                 }
+                if(authId != null) {
+                    o.put("authid", authId);
+                }
+                ArrayNode authMethodsNode = o.putArray("authmethods");
```

Comment:
I didn't put it in the `if(authId != null)` because looking at the specs it's not clear to me whether `authid` and `authmethods` are related for every possible authentication method.

However, checking that `authMethods` is not null and not empty is definitely a better approach than mine ;)